kernel/packages/decentraland-ecs/src/ecs/Entity.ts

```diff
@@ -305,22 +317,23 @@ export class Entity implements IEntity {
       delete currentParent.children[this.uuid]
     }
 
+    // Make sure that the parent and child are both on the engine, or off the engine, together
     if (newParent !== null && newParent.uuid !== '0') {
       if (!newParent.isAddedToEngine() && this.isAddedToEngine()) {
         // tslint:disable-next-line:semicolon
-        ;(this.engine as IEngine).removeEntity(this)
+        (this.engine as IEngine).removeEntity(this)
       }
       if (newParent.isAddedToEngine() && !this.isAddedToEngine()) {
         // tslint:disable-next-line:semicolon
-        ;((newParent as Entity).engine as IEngine).addEntity(this)
+        ((newParent as Entity).engine as IEngine).addEntity(this)
```

Comment:
I know you didn't actually change this code, but we may take the opportunity :)
```suggestion
(newParent as Entity).engine!.addEntity(this)
```